src/mem/PoolMalloc.cc

```diff
@@ -61,39 +62,29 @@ size_t
 MemPoolMalloc::getStats(Mem::PoolStats &stats)
 {
     stats.pool = this;
-    stats.label = objectType();
+    stats.label = label;
     stats.meter = &meter;
-    stats.obj_size = obj_size;
+    stats.obj_size = objectSize;
     stats.chunk_capacity = 0;
 
-    stats.chunks_alloc += 0;
-    stats.chunks_inuse += 0;
-    stats.chunks_partial += 0;
-    stats.chunks_free += 0;
-
     stats.items_alloc += meter.alloc.currentLevel();
     stats.items_inuse += meter.inuse.currentLevel();
     stats.items_idle += meter.idle.currentLevel();
 
-    stats.overhead += sizeof(MemPoolMalloc) + strlen(objectType()) + 1;
-
-    return meter.inuse.currentLevel();
-}
+    stats.overhead += sizeof(*this) + strlen(label) + 1;
 
-int
-MemPoolMalloc::getInUseCount()
-{
-    return meter.inuse.currentLevel();
+    return getInUseCount();
 }
 
-MemPoolMalloc::MemPoolMalloc(char const *aLabel, size_t aSize) : MemImplementingAllocator(aLabel, aSize)
+MemPoolMalloc::MemPoolMalloc(char const *aLabel, size_t aSize) :
+    Mem::Allocator(aLabel, aSize)
 {
 }
 
 MemPoolMalloc::~MemPoolMalloc()
 {
-    assert(meter.inuse.currentLevel() == 0);
     clean(0);
+    assert(getInUseCount() == 0);
```

Comment:
I recommend keeping the old assert()/clean() order. I know you are trying to make the order in two allocator destructors _consistent_, and that is a noble goal, but I would not try to fix this particular inconsistency in _this_ PR because

* Moving misplaced chunking allocator assert to the right position would contradict the existing comments about that destructor. More (potentially controversial) work is needed to clean up that code. We have discussed that in another change request (that has been correctly addressed already by preserving the problematic assert in problematic code).

* Moving this _correctly placed_ assertion makes the code worse -- it is a step backward. This assert is needed not to double check clean() result -- clean() has nothing to do with in-use objects. This assert exists to make sure we do not destruct prematurely (i.e. while we are still needed by some allocated objects). Ideally, it should be separated from the clean() call by an empty line to emphasize that this is an allocator destruction precondition, not a clean() call precondition (but that polishing can be considered lying outside this PR scope).

In summary, it is better to preserve this inconsistency than to make this destructor code worse.

```suggestion
    assert(getInUseCount() == 0);
    clean(0);
```

I do not insist on this change because this whole code area is problematic, the unwanted move in this PR is not going to break anything badly, and I do not have the time to fight for every little improvement in this otherwise useful PR.